src/main/scala/com/databricks/labs/overwatch/utils/PipelineValidationHelper.scala

```diff
@@ -0,0 +1,498 @@
+package com.databricks.labs.overwatch.utils
+
+import com.databricks.labs.overwatch.pipeline.Pipeline
+
+import org.apache.log4j.{Level, Logger}
+import org.apache.spark.sql.functions._
+import com.databricks.labs.overwatch.pipeline._
+import com.databricks.labs.overwatch.pipeline.TransformFunctions._
+import com.databricks.labs.validation._
+import org.apache.spark.sql.DataFrame
+
+import java.time.LocalDateTime
+import scala.collection.mutable.ArrayBuffer
+
+abstract class PipelineValidationHelper(_etlDB: String)  extends SparkSessionWrapper {
+
+  import spark.implicits._
+
+  private val logger: Logger = Logger.getLogger(this.getClass)
+
+  private var _pipelineSnapTime: Long = _
+
+  private val _healthCheck_id: String = java.util.UUID.randomUUID.toString
+  private val _quarantine_id: String = java.util.UUID.randomUUID.toString
+
+  private def healthCheckID: String = _healthCheck_id
+
+  private def quarantineID: String = _quarantine_id
+
+  private def etlDB: String = _etlDB
+
+  private var _validations: ArrayBuffer[HealthCheckReport] = new ArrayBuffer[HealthCheckReport]()
+  private var _quarantine: ArrayBuffer[QuarantineReport] = new ArrayBuffer[QuarantineReport]()
+
+  def validations: ArrayBuffer[HealthCheckReport] = _validations
+
+  def quarantine: ArrayBuffer[QuarantineReport] = _quarantine
+
+  val _isOverwatchDB: Boolean = spark.sessionState.catalog.getDatabaseMetadata(etlDB).properties.getOrElse("OVERWATCHDB", "FALSE").toBoolean
+
+  private def isOverwatchDB = _isOverwatchDB
+
+  private val workSpace = if (isOverwatchDB) {
+    println(s"$etlDB is Overwatch Database and Suitable for Pipeline Validation")
+    Helpers.getWorkspaceByDatabase(etlDB)
+  } else {
+    val errMsg = s"${etlDB} is Not Overwatch Database.Pipeline Validation can only work on Overwatch Database Tables. Validation aborted!!!"
+    throw new BadConfigException(errMsg)
+  }
+
+  private val pipelineDF = if (spark.catalog.tableExists(s"$etlDB.pipeline_report")) {
+    spark.read.table(s"$etlDB.pipeline_report")
+  } else {
+    val errMsg = s"pipeline_report is not present in $etlDB. To proceed with pipeline_validation , pipeline_report table needs to be present in the database.Pipeline Validation aborted!!!"
+    throw new BadConfigException(errMsg)
+  }
+  val storagePrefix: String = pipelineDF.select("inputConfig.dataTarget.etlDataPathPrefix").collect()(0)(0).toString
```

Comment:
you can fetch this value from workspace.getConfig

Reply:
Done